[#4] [Chore] Set up the CD with Fastlane and GitHub Actions for iOS

Thieurom opened the pull request:
- Closes https://github.com/nimblehq/flutter-ic-kaung-thieu/issues/4

## What happened 👀

- Create iOS Staging and Production apps on the AppStore Connect.
- Create profiles for staging and production apps.
- Update code signing settings for Release-staging and Release-production.
- Update minimum iOS to 15.0.
- Update Ruby gems and Cocoa pods.
- Update `fastlane`:
  - Update app identities, constants, and environment variables.
  - Add new lanes to set AppStore Connect API.
  - Update lane: **Build and upload Staging app to Test Flight**.
  - Add new lane: **Build and upload Production app to Test Flight**.
- Create workflows to build and deploy iOS staging and production apps:
  - Set necessary environment variables in GitHub secrets.
  - Cache Ruby gems and Cocoa Pods.

## Insight 📝

- There was a problem with the outdated version of `fastlane` that prevented uploading app build with Xcode 14.
- With `actions/checkout@v3`, need to set `fetch-depth: 0` to fetch all git history, then use number of commits for the build version.

## Proof Of Work 📹

- Workflows ran successfully on GitHub actions, for [deploying staging build](https://github.com/nimblehq/flutter-ic-kaung-thieu/actions/runs/4604910655) and [production build](https://github.com/nimblehq/flutter-ic-kaung-thieu/actions/runs/4604910654).
- Uploaded apps on TestFight, and delivered successfully to Testers:

<img width="1588" alt="Screenshot 2023-04-03 at 15 44 12" src="https://user-images.githubusercontent.com/17972674/229458757-7ed0a41e-677b-4ce3-b03a-6636774cfe2a.png">
<img width="1588" alt="Screenshot 2023-04-03 at 15 44 26" src="https://user-images.githubusercontent.com/17972674/229458779-4aa7a85a-4b16-48e6-bdde-8432306d9416.png">

- Can install and run apps on devices:
  

https://user-images.githubusercontent.com/17972674/229718811-548c02b8-bdd5-4e94-a21d-174169170554.MP4

manh-t commented on ios/Runner.xcodeproj/project.pbxproj:
```diff
@@ -356,7 +353,7 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 15.0;
```
@Thieurom If we set the deployment target to `15.0`, does it mean the user who has iOS 11 device cannot install this app? 👀 
We have the chapter agreement to use `ios 10` as [the minimum version](https://github.com/nimblehq/flutter-templates/issues/37), so I want to know this better 😁

Thieurom replied:
> If we set the deployment target to 15.0, does it mean the user who has iOS 11 device cannot install this app? 👀

That's correct!
I didn't try myself, but it seems there's a problem [version mismatch error while creating ipa for testflight](https://github.com/nimblehq/ic-flutter-taher-toby/pull/51) with `iOS10`.

For the minimum OS version, we should set a lower one on the **flutter-templates** because the purpose of templates is to allow other people to bootstrap a new project quickly. And since we don't know beforehand which OS they target, we try to support a wide range of OS versions. They–the ones who use our templates–then feel free to adjust based on their needs. There's no rule that they must keep exact OS versions in the generated project.

For this specific app, if there's a requirement for the iOS version, it should be indicated in [the IC requirement](https://www.notion.so/nimblehq/Internal-Certification-Path-Mobile-de354e1ce76f4b80922c3db21ea0117f?pvs=4). Since I haven't found such, personally I like to support two latest major OS versions. And for iOS, they are `15.x` and `16.x` 🙏

manh-t replied:
@Thieurom can we set the minimum version to `11` like Taher and Toby IC? 😄 I prefer we can support a wide range of OS version instead of focusing on the latest version 🙏

Thieurom replied:
To be honest, I still don't feel the need to support an iOS version that was released far from 2017. But that's an easy change, and I'll update as per your suggestions 🙏

Thieurom replied:
Updated with https://github.com/nimblehq/flutter-ic-kaung-thieu/pull/37/commits/7bae4871556a52459969caac540a75ffdfcceacb 🙏

manh-t commented on ios/Podfile:
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-platform :ios, '10.0'
+platform :ios, '15.0'
```
same here

manh-t commented on ios/Flutter/AppFrameworkInfo.plist:
```diff
@@ -2,25 +2,25 @@
+	<key>CFBundleVersion</key>
+	<string>1.0</string>
+	<key>MinimumOSVersion</key>
+	<string>15.0</string>
```
and here

manh-t commented on ios/Runner/Info.plist:
```diff
@@ -1,8 +1,9 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>ITSAppUsesNonExemptEncryption</key>
```
@Thieurom can I know what this setting does? 🙏

Thieurom replied:
This is to comply with [Encryption Export Regulations](https://developer.apple.com/documentation/security/complying_with_encryption_export_regulations). Without the compliant, developers can't publish their app to the AppStore and distribute beta versions to testers through TestFlight.

This can be done on the AppStore Connect portal with an `Admin` role. I prefer configuring in code. It's more explicit and easy to trace and update since it's a part of git history.

manh-t replied:
@Thieurom thank you, got it now 🤗

manh-t commented:
@Thieurom can you do a favor by triggering the CI to see whether the minimum supported OS changes doesn't affect the workflow? 🙇

Thieurom commented:
> @Thieurom can you do a favor by triggering the CI to see whether the minimum supported OS changes doesn't affect the workflow? 🙇

@manh-t I triggered the CD to build Staging and Production builds for iOS and updated the link in the PR description ☝️
Also, I added a video showing the apps installed and running on devices 🙏
